
// This file is used to test the wire formatter class.

syntax = "proto3";

enum Test_Enum {
  ZERO  = 0;
  ONE   = 1;
  TWO   = 2;
  ONEHUNDERTTWENTYSEVEN = 127;
  ONEHUNDERTTWENTYEIGHT = 128;
  TWOHUNDERDFIFTYFIVE   = 255;
  TWOBILLION = 2000000000;
}

message Test_Simple_Types {
  int32       a_int32     = 1;
  int64       a_int64     = 2;
  uint32      a_uint32    = 3;
  uint64      a_uint64    = 4; 
  sint32      a_sint32    = 5;
  sint64      a_sint64    = 6; 
  bool        a_bool      = 7;
  Test_Enum   a_enum      = 8;
  fixed64     a_fixed64   = 9;
  sfixed64    a_sfixed64  = 10;
  double      a_double    = 11;
  fixed32     a_fixed32   = 12;
  sfixed32    a_sfixed32  = 13;
  float       a_float     = 14;
}
